server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3310,13 +3331,19 @@ public Vlan dedicatePublicIpRange(final DedicatePublicIpRangeCmd cmd) throws Res
             vlanOwner = _accountMgr.getAccount(project.getProjectAccountId());
         }
 
+        Domain domain = null;
         if (accountName != null && domainId != null) {
             vlanOwner = _accountDao.findActiveAccount(accountName, domainId);
-        }
-        if (vlanOwner == null) {
-            throw new InvalidParameterValueException("Unable to find account by name " + accountName);
-        } else if (vlanOwner.getId() == Account.ACCOUNT_ID_SYSTEM) {
-            throw new InvalidParameterValueException("Please specify a valid account. Cannot dedicate IP range to system account");
+            if (vlanOwner == null) {
```

Comment:
@ustcweizhou This PR is almost ready to be merged. Please respond to comments above.